### PR TITLE
s22-4pm-2 ml - added course fixtures, form, table, create page, and index page and tests

### PR DESCRIPTION
**Pull request by Morgan6688**
# Overview

In this PR, we added the following files in the frontend and wrote tests for all of them:

- courseFixtures
- Course Form
- Course Table
- Course Create Page
- Course Index Page

They all have 100% codecov now and all mutation killed.

# Details
## CodeCov Report
image

## After
Now we have a placeholder:
image

And a table:
image

We can also create a new course now:
image

There's also a delete button so you can delete the course after pressing it:
### Before
image

### After
image

### CODE REVIEW

**Review by scsampath — DISMISSED** [review]
# Course Create
Present on AppNavBar. Created courses accept the correct parameters and appear when listed. Validation is not present, however, when you merge to main it should work!

# Course List
Present on AppNavBar. The courses I created successfully appear in a list. The information is displayed correctly in the table. Courses can also be successfully deleted.
# Tests
All test pass. Frontend line and mutation coverage is 100%.

**Review by scsampath — APPROVED** [review]
# Course Create
Present on AppNavBar. Created courses accept the correct parameters and appear when listed. Validation is not present, however, when you merge to main it should work!

# Course List
Present on AppNavBar. The courses I created successfully appear in a list. The information is displayed correctly in the table. Courses can also be successfully deleted.
# Tests
All test pass. Frontend line and mutation coverage is 100%.

File: frontend/src/main/components/Courses/CourseTable.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        },
+        {
+            Header: 'Personal Schedule ID',
+            accessor: 'psId',
+        },
+    ];
+
+    const columnsIfUser = [
+        ...columns,
+        // ButtonColumn("Edit", "primary", editCallback, "PersonalSchedulesTable"),
```
Comment (pconrad) [inline frontend/src/main/components/Courses/CourseTable.js:37]: Best to remove this until is ready to be implemented.   I'll overlook it if this is the only thing on the code review that needs to be changed.

**Review by pconrad — APPROVED** [review]
LGTM except for the one comment.  But if this clears QA testing, I'll let that slide.

**Review by pconrad — APPROVED** [review]
So... this is a "good start".   

It's not the way we want the UX to be, eventually, but I think we can merge it since it gets us closer to what we want.

A few things:

When the enrollCd is invalid, right now we get a weird toast backend error, instead of a "nice" error message.   We should probably intercept the backend error message and use it to put up a nicer error message that will make sense to the user.

For now, the "global"  List Courses table is fine, but it isn't what we want.  What we want is that the courses that belong to each personal schedule appear on the personal schedule details page.

I.e. where it says: "here is where the courses detail goes"

image


Instead of entering a PSId, you should move the "Add Course" from the menu to an "Add Course" button on the PS Details page.     Better yet, embed the form to add a new course DIRECTLY on the PS details page, and use the PSId for that page.  So adding a course is simple entering an enrollCd and pressing enter.

**Pull request merged**